File: board_dsl/src/main/java/com/dooho/board/repository/CommentRepository.java
package com.dooho.board.repository;

import com.dooho.board.entity.CommentEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Repository
@Transactional
public interface CommentRepository extends JpaRepository<CommentEntity, Integer> {
    List<CommentEntity> findByBoardNumberOrderByCommentWriteDate(Integer boardNumber);

    Integer countByBoardNumber(Integer boardNumber);

    void deleteByCommentId(Integer commentId);

    List<CommentEntity> findByUserEmail(String userEmail);
}

File: board_spring/src/main/java/com/dooho/board/repository/CommentRepository.kt
package com.dooho.board.repository

import com.dooho.board.entity.CommentEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
@Transactional
interface CommentRepository : JpaRepository<CommentEntity?, Int?> {
    fun findByBoardNumberOrderByCommentWriteDate(boardNumber: Int?): List<CommentEntity?>?
    fun countByBoardNumber(boardNumber: Int?): Int?
    fun deleteByCommentId(commentId: Int?)
    fun findByUserEmail(userEmail: String?): List<CommentEntity?>?
}